### PR TITLE
Adding Cilium to create_stack

### PR DESCRIPTION
**Pull request by ahsan518**
This change adds Cilium to create_stack script, and uses crio
by default unless the user otherwise speicify

Signed-Off: Syed Ahsan <syed.ahsan.shamim.zaidi@intel.com>

#179

### CODE REVIEW

File: clr-k8s-examples/create_stack.sh [hunk 1]
```diff
@@ -16,10 +16,13 @@ HIGH_POD_COUNT=${HIGH_POD_COUNT:-""}
 
 # versions
 CANAL_VER="${CLRK8S_CANAL_VER:-v3.9}"
+CILIUM_VER="${CLRK8S_CILIUM_VER:-v1.6}"
 K8S_VER="${CLRK8S_K8S_VER:-}"
 ROOK_VER="${CLRK8S_ROOK_VER:-v1.1.1}"
 METRICS_VER="${CLRK8S_METRICS_VER:-v0.3.5}"
 PROMETHEUS_VER="${CLRK8S_PROMETHEUS_VER:-f458e85e5d7675f7bc253072e1b4c8892b51af0f}"
+CNI=${CNI:-"canal"}
```
Comment (jascott1) [inline clr-k8s-examples/create_stack.sh]: Can you take the CNI external var as CLRK8S_CNI? I would like the same for RUNNER (CLRK8S_RUNNER) but we already have it external to the script as RUNNER. Perhaps we can fix that one in the future as well.

File: clr-k8s-examples/create_stack.sh [hunk 2]
```diff
@@ -96,23 +99,45 @@ function kata() {
...
+        	kubectl apply -k "${CANAL_DIR}/overlays/${CANAL_VER}" --validate=false
+		;;
+	cilium)
+		CILIUM_VER=${1:-$CILIUM_VER}
+		CILIUM_URL="https://github.com/cilium/cilium/archive/$CILIUM_VER.tar.gz"
+		CILIUM_DIR="0-cilium"
+
+		mkdir -p "${CILIUM_DIR}/overlays/${CILIUM_VER}/cilium"
+		pushd "${CILIUM_DIR}/overlays/${CILIUM_VER}/cilium/"
+		wget -O "$CILIUM_VER.tar.gz" "$CILIUM_URL"
```
Comment (jascott1) [inline clr-k8s-examples/create_stack.sh]: Can we use git here?

File: clr-k8s-examples/create_stack.sh [hunk 3]
```diff
@@ -96,23 +99,45 @@ function kata() {
...
+		tar xzf "$CILIUM_VER.tar.gz"
+		pushd cilium-"${CILIUM_VER#*v}"/install/kubernetes
+		helm template cilium --namespace kube-system --set global.containerRuntime.integration="$RUNNER" >cilium.yaml
```
Comment (jascott1) [inline clr-k8s-examples/create_stack.sh]: Can we use kustomize?

File: clr-k8s-examples/create_stack.sh (hunk 1 above)
Comment (knangia) [reply clr-k8s-examples/create_stack.sh]: If we are making "CLRK8S_RUNNER" as environment variable, then we probably should update it throughout the repo. I can create a PR for the same, if its priority. @jascott1

File: clr-k8s-examples/create_stack.sh [hunk 4]
```diff
@@ -16,10 +16,13 @@ HIGH_POD_COUNT=${HIGH_POD_COUNT:-""}
 
 # versions
 CANAL_VER="${CLRK8S_CANAL_VER:-v3.9}"
+CILIUM_VER="${CLRK8S_CILIUM_VER:-v1.6}"
 K8S_VER="${CLRK8S_K8S_VER:-}"
 ROOK_VER="${CLRK8S_ROOK_VER:-v1.1.1}"
 METRICS_VER="${CLRK8S_METRICS_VER:-v0.3.5}"
 PROMETHEUS_VER="${CLRK8S_PROMETHEUS_VER:-f458e85e5d7675f7bc253072e1b4c8892b51af0f}"
+CLRK8S_CNI=${CLRK8S_CNI:-"canal"}
```
Comment (jascott1) [inline clr-k8s-examples/create_stack.sh]: nit: as a pattern we are stripping the CLRK8S for internal vars. `CNI=${CLRK8S_CNI:-"canal"}`

File: clr-k8s-examples/create_stack.sh [hunk 5]
```diff
@@ -96,23 +99,40 @@ function kata() {
...
+		;;
+	cilium)
+		CILIUM_VER=${1:-$CILIUM_VER}
+		CILIUM_URL="https://github.com/cilium/cilium.git"
+		CILIUM_DIR="0-cilium"
+
+		mkdir -p "${CILIUM_DIR}/overlays/${CILIUM_VER}"
+		get_repo "${CILIUM_URL}" "${CILIUM_DIR}/overlays/${CILIUM_VER}"
+		set_repo_version "${CILIUM_VER}" "${CILIUM_DIR}/overlays/${CILIUM_VER}/cilium/"
+		helm template "${CILIUM_DIR}/overlays/${CILIUM_VER}/cilium/install/kubernetes/cilium" --namespace kube-system --set global.containerRuntime.integration="$CLRK8S_RUNNER" | kubectl apply -f -
```
Comment (jascott1) [inline clr-k8s-examples/create_stack.sh]: CLRK8SRUNNER should be RUNNER now. Throws unbound variable.

File: clr-k8s-examples/create_stack.sh [hunk 6]
```diff
@@ -96,23 +99,40 @@ function kata() {
...
+		CILIUM_URL="https://github.com/cilium/cilium.git"
+		CILIUM_DIR="0-cilium"
+
+		mkdir -p "${CILIUM_DIR}/overlays/${CILIUM_VER}"
+		get_repo "${CILIUM_URL}" "${CILIUM_DIR}/overlays/${CILIUM_VER}"
+		set_repo_version "${CILIUM_VER}" "${CILIUM_DIR}/overlays/${CILIUM_VER}/cilium/"
+		helm template "${CILIUM_DIR}/overlays/${CILIUM_VER}/cilium/install/kubernetes/cilium" --namespace kube-system --set global.containerRuntime.integration="$RUNNER" | kubectl apply -f -
+		;;
+	*)
+		echo"Unknown cni $CLRK8S_CNI"
```
Comment (jascott1) [inline clr-k8s-examples/create_stack.sh]: Remove the `CLRK8S_` as its is causing undeclared var error in bash.

File: clr-k8s-examples/create_stack.sh [hunk 7]
```diff
@@ -96,23 +99,40 @@ function kata() {
...
-
-	# canal manifests are not kept in repo but in docs site so use curl
-	mkdir -p "${CANAL_DIR}/overlays/${CANAL_VER}/canal"
-	curl -o "${CANAL_DIR}/overlays/${CANAL_VER}/canal/canal.yaml" "$CANAL_URL/canal.yaml"
-	if [[ "$CANAL_VER" == "v3.3" ]]; then
-		curl -o "${CANAL_DIR}/overlays/${CANAL_VER}/canal/rbac.yaml" "$CANAL_URL/rbac.yaml"
-	fi
-	# canal doesnt pass kustomize validation
-	kubectl apply -k "${CANAL_DIR}/overlays/${CANAL_VER}" --validate=false
+	case "$CLRK8S_CNI" in
```
Comment (jascott1) [inline clr-k8s-examples/create_stack.sh]: Remove the CLRK8S_ as its is causing undeclared var error in bash.

Comment (ahsan518) [reply clr-k8s-examples/create_stack.sh]: Done

**Comment by CraigSterrett** [comment]
Testing passed on Vagrant  running create_stack all and even passed sonobuoy

**Comment by jascott1** [comment]
@ahsan518 can you rebase this right quick?

**Comment by jascott1** [comment]
Note: This is only working on Vagrant and not on OpenStack. A bug will be filed to track progress for OpenStack.

**Pull request merged**